### PR TITLE
Added feature: Show user info by pressing 'i'

### LINKED ISSUE #511: Feature: Enable view of user info
Just like the stream descriptions, on pressing `i`, while a user is in focus, the full description of the user should pop-up which must include the information available in the `View full profile` section on the web app.

### PR DESCRIPTION
**Pull request by SGeetansh**
Fixes #511 
This pull request implements a feature of displaying the information of the user in focus. All the information that is visible in the web app is present.
It doesn't pass the tests right now. Please suggest any required changes.

### CODE REVIEW

File: zulipterminal/config/keys.py [hunk 1]
```diff
@@ -200,6 +200,11 @@
         'help_text': 'View stream description',
         'key_category': 'stream_list',
     }),
+    ('USER_INFO', {
```
Comment (sumanthvrao) [inline zulipterminal/config/keys.py]: It would also benefit to have a note for this in `README`. (Maybe these two changes can be extracted into a separate commit)

File: zulipterminal/model.py [hunk 2]
```diff
@@ -79,14 +79,14 @@ def __init__(self, controller: Any) -> None:
         self.stream_id = -1
         self.recipients = frozenset()  # type: FrozenSet[Any]
         self.index = initial_index
-
+
```
Comment (sumanthvrao) [inline zulipterminal/model.py]: Some extra spaces ;)

File: zulipterminal/ui_tools/buttons.py [hunk 3]
```diff
@@ -213,6 +215,11 @@ def _narrow_with_compose(self, button: Any) -> None:
         self._view.body.focus.original_widget.set_focus('footer')
         self._view.write_box.private_box_view(self)
 
+    def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
+        if is_command_key('USER_INFO', key):
+            self.model.controller.show_user_info(self.user_id)
```
Comment (sumanthvrao) [inline zulipterminal/ui_tools/buttons.py]: alternatively, can we could just use `self.controller.show_user_info` here , and avoid the line on top? :)

File: zulipterminal/core.py [hunk 4]
```diff
@@ -120,6 +122,65 @@ def show_stream_info(self, color: str, name: str, desc: str) -> None:
         show_stream_view = StreamInfoView(self, color, name, desc)
         self.show_pop_up(show_stream_view, "# {}".format(name))
 
+    def show_user_info(self, user_id:int) -> None:
+        user_id = user_id
+        url = 'users/' + str(user_id)
+        result = self.client.call_endpoint(
```
Comment (sumanthvrao) [inline zulipterminal/core.py]: It's always good to check if the result was a success or not here. We do that in many places so you can have a look.

Comment (SGeetansh) [reply zulipterminal/core.py]: I checked the other places. They just check if the response message was 'success' or not, with a return statement at the end of the function (`return response['result'] == success`). Do you want me to just write that only? Sorry, if I am wrong, but can you please elaborate on this? I didn't put a try-except here because I think it isn't required.

Comment (sumanthvrao) [reply zulipterminal/core.py]: Yup, that's the correct way to do the validation. No need of a try-catch block.

File: zulipterminal/config/keys.py [hunk 5]
```diff
@@ -200,6 +200,11 @@
...
+        'keys': {'i'},
```
Comment (sumanthvrao) [inline zulipterminal/config/keys.py]: You could extract the README and keys into a sepearte commit as well. As it stands, this commit has a lot of file changes.

File: requirements.txt [hunk 6]
```diff
@@ -13,5 +13,4 @@ emoji==0.5.0
 urwid_readline==0.10
 beautifulsoup4==4.6.0
 lxml==4.2.3
-
-zipp==1.0.0  # To support Python 3.5
+zipp==1.0.0  # To support Python 3.5
```
Comment (sumanthvrao) [inline requirements.txt]: Unwanted diff? :)

File: zulipterminal/ui_tools/views.py [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import threading
 import time
+import math
```
Comment (sumanthvrao) [inline zulipterminal/ui_tools/views.py]: Isn't `import math` used in your third commit. If so, let's import it only in the commit where it is used.

File: zulipterminal/core.py [hunk 8]
```diff
@@ -120,9 +122,64 @@ def show_stream_info(self, color: str, name: str, desc: str) -> None:
...
+        if res_data['is_admin']:
+            display_data['Role'] = "Admin"
+        elif res_data['is_guest']:
+            display_data['Role'] = "Guest"
+        else:
+            display_data['Role'] = "Member"
+
+        show_userinfo_view = UserInfoView(self, display_data)
+        self.show_pop_up(show_userinfo_view, "{}".format(res_data['full_name']))
+        return response['result'] == 'success'
```
Comment (sumanthvrao) [inline zulipterminal/core.py]: Where do you validate whether the return value was true/false?

File: zulipterminal/ui_tools/views.py [hunk 9]
```diff
@@ -850,13 +850,26 @@ def keypress(self, size: Tuple[int, int], key: str) -> str:
 
 
 class UserInfoView(urwid.ListBox):
-    def __init__(self, controller, display_data) -> None:
+    def __init__(self, controller, display_data:dict) -> None:
```
Comment (sumanthvrao) [inline zulipterminal/ui_tools/views.py]: Maybe type-hint more?

File: zulipterminal/core.py (hunk 8 above)
Comment (SGeetansh) [reply zulipterminal/core.py]: Can you point me to an instance where this is done? I don't know what to do.

Comment (sumanthvrao) [reply zulipterminal/core.py]: @SGeetansh I meant you could do this at the beginning of this function after you make the API call. Only of the response result is a success, you can parse and store the parameters as you have done.

**Comment by neiljp** [comment]
@SGeetansh I hope you had a good Summer - just checking in if you planned to update this PR?

If you do plan to update this:
* Some timezone code has been added, so the dependency commit shouldn't be necessary now
* The popup code has been streamlined, which may well make this easier - see `PopUpView` and child classes
* Just having the available information from the existing presences data would be good in a working commit, then following up with custom profile fields in a later one - the titles need sourcing from somewhere as I think the values you're using are hard-coded from chat.zulip.org?

**Comment by neiljp** [comment]
Closing, as this was replaced via #848.

@SGeetansh Thanks for filing the issue and taking a shot here :+1:
